To: vim_dev@googlegroups.com
Subject: Patch 7.3.067
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.067 (after 7.3.058)
Problem:    Ruby: Init_prelude is not always available.
Solution:   Remove use of Init_prelude. (Yasuhiro Matsumoto)
Files:	    src/if_ruby.c


*** ../vim-7.3.066/src/if_ruby.c	2010-11-16 14:46:14.000000000 +0100
--- src/if_ruby.c	2010-11-24 16:53:06.000000000 +0100
***************
*** 231,237 ****
  # define rb_enc_str_new			dll_rb_enc_str_new
  # define rb_intern2			dll_rb_intern2
  # define rb_const_remove		dll_rb_const_remove
- # define Init_prelude			dll_Init_prelude
  # define rb_sprintf			dll_rb_sprintf
  # define ruby_init_stack		dll_ruby_init_stack
  #endif
--- 231,236 ----
***************
*** 433,439 ****
      {"rb_enc_str_new", (RUBY_PROC*)&dll_rb_enc_str_new},
      {"rb_intern2", (RUBY_PROC*)&dll_rb_intern2},
      {"rb_const_remove", (RUBY_PROC*)&dll_rb_const_remove},
-     {"Init_prelude", (RUBY_PROC*)&dll_Init_prelude},
      {"rb_sprintf", (RUBY_PROC*)&dll_rb_sprintf},
      {"ruby_init_stack", (RUBY_PROC*)&dll_ruby_init_stack},
  #endif
--- 432,437 ----
***************
*** 675,681 ****
  	    /* This avoids the error "Encoding::ConverterNotFoundError: code
  	     * converter not found (UTF-16LE to ASCII-8BIT)". */
  	    rb_define_module("Gem");
- 	    Init_prelude();
  	    rb_const_remove(rb_cObject, rb_intern2("TMP_RUBY_PREFIX", 15));
  #endif
  	    ruby_vim_init();
--- 673,678 ----
*** ../vim-7.3.066/src/version.c	2010-11-24 16:31:55.000000000 +0100
--- src/version.c	2010-11-24 16:53:35.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     67,
  /**/

-- 
CART DRIVER: Bring out your dead!
LARGE MAN:   Here's one!
CART DRIVER: Ninepence.
BODY:        I'm not dead!
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
